web/reporting/views.py

```diff
@@ -1,17 +1,43 @@
+import csv
 import dataclasses
+from datetime import date
+from io import StringIO
 
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import *
 from django.contrib.admin.views.decorators import staff_member_required
-from django.http import HttpRequest, JsonResponse
+from django.db import connection
+from django.http import HttpRequest, HttpResponse, JsonResponse
 
-from reporting.matomo import usage
 from main.test.test_permissions_helpers import no_perms_test
+from reporting.matomo import usage
 
 from .admin.usage_dashboard import DateForm
 
 
+def sql_to_csv_response(sql: str, filename: str) -> HttpResponse:
```

Comment:
I'm sure this isn't necessary, but it crossed my mind after a frisson of horror on reading "Given a blob of SQL, execute it": should `sql_to_csv_response()` ensure that it can only run select queries?

Reply:
Nah, it doesn't make sense -- I'm just twitchy.

Reply:
I'm glad you asked!